application/src/main/java/org/togetherjava/tjbot/features/moderation/history/PurgeMessageListener.java

```diff
@@ -0,0 +1,118 @@
+package org.togetherjava.tjbot.features.moderation.history;
+
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import org.jooq.impl.UpdatableRecordImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.records.MessageHistoryRecord;
+import org.togetherjava.tjbot.features.MessageReceiverAdapter;
+
+import java.time.Instant;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import static org.togetherjava.tjbot.db.generated.Tables.MESSAGE_HISTORY;
+
+/**
+ * Listens for new messages throughout the guild and stores metadata for each message in a database.
+ * <p>
+ * Note: This class also trims database records if they reach a maximum records limit.
+ * </p>
+ */
+public final class PurgeMessageListener extends MessageReceiverAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(PurgeMessageListener.class);
+    private static final int MESSAGES_RECORDS_MAX_LIMIT = 7500;
+    private static final int MESSAGES_RECORDS_THRESHOLD = MESSAGES_RECORDS_MAX_LIMIT - 100;
+    private static final int RECORDS_TO_TRIM = 500;
+    static AtomicInteger recordsCounter = new AtomicInteger(0);
```

Comment:
since the PR is getting too old now and so much stuff to review , flagging the important bit, is using atomic justified here?
the goal is to limit only certain no of db entries(recordsCounter), which is modified across package in `PurgeMessageListener` (incrementCounterByOne) on each new message and in both `PurgeHistoryCommand` and `PurgeExpiredMessageHistory` (decrementCountByOne) on each deletion of record/message.